health/healthstate.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package health
+
+import (
+	"tailscale.com/util/mak"
+	"tailscale.com/version"
+)
+
+// HealthState is a representation of the health status of the backend, which is
+// provided to the client UI via LocalAPI. It is used to simplify the data structure
+// provided over JSON.
+type HealthState struct {
+	// Warnings maps each Warnable code to a map of WarningCode to Warning
+	Warnings map[WarnableCode]map[WarningCode]*Warning
+	// DebugMode is whether we are in debug mode. When in debug mode, the GUI will
+	// ignore severity levels, and display all Warnings with the same prominence.
+	DebugMode bool
+}
+
+func (t *Tracker) CurrentHealthState() *HealthState {
+	if t.nil() {
+		return nil
+	}
+
+	t.mu.Lock()
+	defer t.mu.Unlock()
+
+	warningsMap := map[WarnableCode]map[WarningCode]*Warning{}
+
+	for _, w := range t.warnables {
+		if warnings, ok := t.warnableVal[w]; ok {
+			mak.Set(&warningsMap, w.Code, warnings)
+		}
+	}
+
+	return &HealthState{
+		Warnings:  warningsMap,
+		DebugMode: version.IsUnstableBuild(), // TODO: probably should live in ipn.Prefs
```

Comment:
probably drop this. make it opt in via envknob then make it a per-Warnable field saying whether it only fires if people opt in to it.

Reply:
done